keras/engine/data_adapter.py

```diff
@@ -1308,6 +1314,9 @@ def __init__(
             strategy, x, steps_per_epoch, class_weight, distribute
         )
 
+        if self._inferred_steps == 0:
+            raise ValueError("Expected input data to `fit()` to be non-empty.")
```

Comment:
Note that this method can be called for model.eval/predict as well, so let's remove the model.fit() from the message.